contracts/ContinuousTokenFactory.sol

```diff
@@ -2,9 +2,193 @@
 
 pragma solidity >=0.6.0 <0.7.0;
 
+import "@openzeppelin/contracts/math/SafeMath.sol";
 import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
-import "@openzeppelin/contracts/access/Ownable.sol";
+import "./ERC20Managed.sol";
+import "./Curve.sol";
 
 contract ContinuousTokenFactory {
+  using SafeMath for uint256;
 
+  ERC20 public reserveToken;
+
+  struct ContinuousToken {
+    address token;
+    address curve;
+  }
+
+  mapping(bytes32 => ContinuousToken) public continuousTokens;
+
+  constructor(address _reserveToken) public {
+    require(
+      _reserveToken != address(0),
+      "ContinuousTokenFactory: no reserve token ERC20 address"
+    );
+    reserveToken = ERC20(_reserveToken);
+  }
+
+  event NewContinuousToken(
+    bytes32 id,
+    address token,
+    string name,
+    address curve
+  );
+
+  function newContinuousToken(
+    bytes32 id, 
+    string memory name, 
+    string memory symbol,
+    address curve
+  ) public returns (bool) {
+    require(!exists(id), "ContinuousTokenFactory: invalid id");
+    // require(Curve(curve).price(0, 1), "ContinuousTokenFactory: invalid curve");
+
+    address token = address(new ERC20Managed(name, symbol));
+    ContinuousToken memory continuousToken = ContinuousToken(token, curve);
+    continuousTokens[id] = continuousToken;
+
+    emit NewContinuousToken(
+      id,
+      token,
+      name,
+      curve
+    );
+    return true;
+  }
+
+  event Buy(
+    bytes32 id, 
+    string name,
+    uint256 amount,
+    uint256 price,
+    address buyer
+  );
+
+  function buy(
+    bytes32 id, 
+    uint256 amount,
+    uint256 maxPrice,
+    address buyer
+  ) external returns (bool) {
+    (ERC20Managed token, Curve curve) = unpack(id);
+
+    // Check price
+    uint256 price = curve.price(token.totalSupply(), amount);
+    require(price <= maxPrice, "ContinuousTokenFactory: price greater than maxPrice");
+
+    // Transfer and mint
+    require(
+      reserveToken.transferFrom(buyer, address(this), price)
+    );
+    token.mint(buyer, amount);
+
+    emit Buy(
+      id,
+      token.name(),
+      amount,
+      price,
+      buyer
+    );
+    return true;
+  }
+
+  event Sell(
+    bytes32 id,
+    string name,
+    uint256 amount,
+    uint256 price,
+    address seller
+  );
+
+  function sell(
+    bytes32 id,
+    uint256 amount,
+    uint256 minPrice,
```

Comment:
I wonder if `price` is the right term for what you get back from a sale. I guess it is, though I was using `minValue`. Could see arguments for `minProceeds` too. (though Price is probably fine)